sound/soc/intel/boards/sof_nau8825.c

```diff
@@ -297,14 +297,6 @@ static int sof_audio_probe(struct platform_device *pdev)
 }
 
 static const struct platform_device_id board_ids[] = {
-	{
-		.name = "sof_nau8825",
-		.driver_data = (kernel_ulong_t)(SOF_SSP_PORT_CODEC(0) |
-					SOF_NUM_IDISP_HDMI(4) |
-					SOF_SSP_PORT_BT_OFFLOAD(2) |
-					SOF_BT_OFFLOAD_PRESENT),
```

Comment:
`adl_nau8825_def` has `SOF_SSP_PORT_AMP(1)` which is not in `sof_nau8825`. Is it matter?

Reply:
It doesn't matter. Machine driver now will detect the existence/type of headphone codec and speaker amplifier before creating DAI link array. SOF_SSP_PORT_AMP(1) will be ignored if there is no amplifier on the board.